python/paddle/fluid/tests/unittests/op_test.py

```diff
@@ -1776,3 +1776,16 @@ def _get_gradient(self,
                              fetch_list,
                              scope=scope,
                              return_numpy=False)))
+
+
+class OpTestTool:
+    @classmethod
+    def skipIf(cls, condition: object, reason: str):
+        return unittest.skipIf(condition, reason)
+
+    @classmethod
+    def skipIfNotBF16(cls):
+        return OpTestTool.skipIf(
+            not (isinstance(_current_expected_place(), core.CPUPlace) and
```

Comment:
Shouldn't it be like: skipIf(not CPU or not SupportsBF16)?

Reply:
We want the test to be run with both CPU and BF16 support, otherwise (logical not) skip.

Reply:
My bad, we meant the same thing but I didn't see one parenthesis

Reply:
Hi, I have two questions:
1. It looks 
```
    @classmethod
    def skipIf(cls, condition: object, reason: str):
        return unittest.skipIf(condition, reason)
```
is not really needed, why don't you call this unittest.skipIf in skipIfNotBF16 ?
2. Why do you create a new class OpTestTool, why don't you just put this class decorator function `def skipIfNotBF16(cls):` in OpTest ?

Reply:
`OpTestTool` was intended to be namespace for convenient toolbox. It contains classmethods that can be used as decorators which gives flexibility we need.

Reply:
Got explanation that keeping following in OpTestTool because we make all possible skip in one class, it will be easy to use and maintain.
```
@classmethod
    def skipIf(cls, condition: object, reason: str):
```
Thanks !